Sources/Sentry/Public/SentryEnvelopeItemHeader.h

```diff
@@ -0,0 +1,25 @@
+#import "SentrySerializable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryEnvelopeItemHeader : NSObject <SentrySerializable>
```

Comment:
`h`: Does this need to be public?

Reply:
Well. SentryEnvelope is public and it have some APIs that requires `SentryEnvelopeItem`, that has public API that requires `SentryEnvelopeItemHeader`.

Maybe those APIs don't need to be public. But this requires another analysis. 
Maybe another PR.

Reply:
Yes, SentryEnvelope doesn't need to be public. Let's solve this in another issue: https://github.com/getsentry/sentry-cocoa/issues/2542